From 9d610ad69937755ea34fb85d4bc85975ead7d396 Mon Sep 17 00:00:00 2001
From: Julian Greilich <j.greilich@gmx.de>
Date: Thu, 9 Nov 2023 14:15:43 +0100
Subject: Silence expected java exceptions

Since we expect the calls to "connect" and "close" in
checkIsTargetLost() to fail with IOException when the target was lost,
we dont need to print the exceptions as log warning.

This restores the behavior like it was before the port to the new Qt 6
Android APIs (7d2c6ae66fd61be56b35ae0c59354b695a02d6c7).

Pick-to: 6.6 6.5
Change-Id: Ic9e14a58827f8b24865202012708644b903d5d70
Reviewed-by: Timur Pocheptsov <timur.pocheptsov@qt.io>
(cherry picked from commit 0692168870463c7f32434b5ee1b0e6d762e078fa)
---
 src/nfc/qnearfieldtarget_android.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git x/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp y/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
index 63e7d62b9f0980762464083cf915765abd8f18e7..c7f842ae781bd0e24865a1bf011af2accd8dd739 100644
--- x/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
+++ y/qtconnectivity/src/nfc/qnearfieldtarget_android.cpp
@@ -304,14 +304,14 @@ void QNearFieldTargetPrivateImpl::checkIsTargetLost()
     methodId = env.findMethod<void>(tagTech.objectClass(), "connect");
     if (methodId)
         env->CallVoidMethod(tagTech.object(), methodId);
-    if (!methodId || env.checkAndClearExceptions()) {
+    if (!methodId || env.checkAndClearExceptions(QJniEnvironment::OutputMode::Silent)) {
         handleTargetLost();
         return;
     }
     methodId = env.findMethod<void>(tagTech.objectClass(), "close");
     if (methodId)
         env->CallVoidMethod(tagTech.object(), methodId);
-    if (!methodId || env.checkAndClearExceptions())
+    if (!methodId || env.checkAndClearExceptions(QJniEnvironment::OutputMode::Silent))
         handleTargetLost();
 }
 
